tests/common/query_check.rs

```diff
@@ -0,0 +1,7 @@
+pub fn query_check<T: lal::Backend>(env_name: &str, backend: &T) {
+    let r = lal::query(backend, Some(env_name), "hello", false);
+    assert!(r.is_ok(), "could query for hello");
+
+    let rl = lal::query(backend, Some(env_name), "hello", true);
+    assert!(rl.is_ok(), "could query latest for hello");
+}
```

Comment:
a lot of tiny files with very specific names after this refactor.

maybe we should make some of these functions match the subcommand we're testing.
e.g. this file contains all the tests for query?

Reply:
Yes. That's the plan.

Any thoughts about how to do this?
Or where to make the splits?

The problem I'm having is that most commands depend on other commands. (e.g. build probably depends on fetch, everything depends on configure).

Reply:
Okay, I've think I've got this now.

Each command gets it's own `test/cases/test_*.rs` file.
resolving this comment.